scripts/test_examples.sh

```diff
@@ -201,43 +212,255 @@ stopKafka
+")
+
+cmd="examples/kafka-plugin/manual-partitioner/manual-partitioner --kafka-config examples/kafka-plugin/manual-partitioner/kafka.conf --messageCount 1 --offsetMsg 8"
+testOutput "${cmd}" "${expected}" "${unexpected}"
+
+# Let us test - in example no new messages generated - we want to list all latest messages
+expected=("Offset: -1, message count: 0
+All plugins initialized successfully
+Sending 0 sync Kafka notifications (protobuf) ...
+Sending 0 async Kafka notifications (protobuf) ...
+Sync watcher closed
+Async watcher closed
+")
+
+unexpected=("Error while stopping watcher
+")
+
+# this is not working  cmd="examples/kafka-plugin/manual-partitioner/manual-partitioner --kafka-config examples/kafka-plugin/manual-partitioner/kafka.conf --messageCount 0 -offsetMsg=\"latest\""
+cmd="examples/kafka-plugin/manual-partitioner/manual-partitioner --kafka-config examples/kafka-plugin/manual-partitioner/kafka.conf --messageCount 0 -offsetMsg=latest"
+testOutput "${cmd}" "${expected}" "${unexpected}"
+
+
+# Let us test - in example no new messages generated - we want to list all oldest messages
+expected=("Offset: -2, message count: 0
+All plugins initialized successfully
+Received async Kafka Message, topic 'example-async-topic', partition '2', offset '0', key: 'async-proto-key', 
+Received async Kafka Message, topic 'example-async-topic', partition '2', offset '11', key: 'async-proto-key', 
+Received sync Kafka Message, topic 'example-sync-topic', partition '1', offset '0', key: 'proto-key', 
+Received sync Kafka Message, topic 'example-sync-topic', partition '1', offset '11', key: 'proto-key', 
+Sending 0 sync Kafka notifications (protobuf) ...
+Sending 0 async Kafka notifications (protobuf) ...
+Sync watcher closed
+Async watcher closed
+")
+
+unexpected=("Error while stopping watcher
+")
+
+cmd="examples/kafka-plugin/manual-partitioner/manual-partitioner --kafka-config examples/kafka-plugin/manual-partitioner/kafka.conf --messageCount 0 -offsetMsg=oldest"
+testOutput "${cmd}" "${expected}" "${unexpected}"
+
+
+# Let us test - in example no new messages generated - wrong value of parameter offsetMsg
+expected=("
+Error loading core: plugin KafkaExample: Init error 'incorrect sync offset value wronginput
+")
+
+unexpected=("Error while stopping watcher
+")
+
+cmd="examples/kafka-plugin/manual-partitioner/manual-partitioner --kafka-config examples/kafka-plugin/manual-partitioner/kafka.conf --messageCount 0 -offsetMsg=wronginput"
+testOutput "${cmd}" "${expected}" "${unexpected}"
+
 
 stopKafka
 
 #### Kafka-plugin hash-partitioner #######################################
 
 startKafka
 
-expected=("Sending 10 sync Kafka notifications
-Sending 10 async Kafka notifications
-Received Kafka Message, topic 'example-sync-topic'
-Async message successfully delivered, topic 'example-async-topic'
-Received async Kafka Message, topic 'example-async-topic'
+# Let us test the running without parameters - in example are generated 10 Kafka Messages to both topics
```

Comment:
in every hash partitioner example - consider making of partition number wildcarded. Hash partitioner is choosing the partitions according to the key (async-proto-key for example, that means all sync messages should be on the same partition and all async messages should be on a different partition). Although partition numbers are the same in most cases, there can be rare cases when the partition numbers will be different and the test will fail

Reply:
At the present time the hash-partitioner example uses kafka docker started with unchanged server.properties where there is the setting:
# The default number of log partitions per topic
num.partitions=1

So maybe it could be better to add the other test case where num.partitions per topic is 3 and in this case it makes sense to make partition number wildcarded?!